app/src/app/[lng]/data/[step]/page.tsx

```diff
@@ -285,87 +285,104 @@ export default function AddDataSteps({
-                borderWidth={2}
-                className="shadow-none hover:drop-shadow-xl transition-shadow"
-              >
-                <Icon as={source.icon} boxSize={9} mb={6} />
-                <Heading size="sm" noOfLines={2}>
-                  {source.title}
-                </Heading>
-                <Flex direction="row" my={4}>
-                  <Tag mr={1}>
-                    <TagLeftIcon
-                      as={MdPlaylistAddCheck}
-                      boxSize={4}
-                      color="content.tertiary"
-                    />
-                    <TagLabel fontSize={12}>
-                      {t("data-quality")}: {t("quality-" + source.dataQuality)}
-                    </TagLabel>
-                  </Tag>
-                  <Tag>
-                    <TagLeftIcon
-                      as={FiTarget}
-                      boxSize={4}
-                      color="content.tertiary"
-                    />
-                    <TagLabel fontSize={12}>
-                      {t("scope")}: {source.scopes.join(", ")}
-                    </TagLabel>
-                  </Tag>
-                </Flex>
-                <Text color="content.tertiary" noOfLines={5}>
-                  {source.description}
-                </Text>
-                <Link className="underline" mt={4} mb={6}>
-                  {t("see-more-details")}
-                </Link>
-                {source.isConnected ? (
-                  <Button
-                    variant="solidPrimary"
-                    px={6}
-                    py={4}
-                    leftIcon={<Icon as={MdCheckCircle} />}
-                  >
-                    {t("data-connected")}
-                  </Button>
-                ) : (
-                  <Button
-                    variant="outline"
-                    bgColor="background.neutral"
-                    onClick={() => onConnectClick(source)}
-                  >
-                    {t("connect-data")}
-                  </Button>
-                )}
-              </Card>
-            ))}
+          {areDataSourcesLoading ? (
+            <Center>
+              <Spinner size="lg" />
+            </Center>
+          ) : !dataSources ? (
+            <Center>
+              <WarningIcon boxSize={8} color="semantic.danger" />
+            </Center>
+          ) : (
+            dataSources
+              .slice(0, isDataSectionExpanded ? dataSources.length : 6)
+              .map((source) => (
+                <Card
+                  key={source.datasourceId}
+                  onClick={() => onSourceClick(source)}
+                  variant="outline"
+                  borderColor={
+                    (source.isConnected && "interactive.tertiary") || undefined
+                  }
+                  borderWidth={2}
+                  className="shadow-none hover:drop-shadow-xl transition-shadow"
+                >
+                  {/* TODO add icon to DataSource */}
```

Comment:
We probably won't require this here right?

Reply:
Well it's mainly a reminder to go back in and add the icon here once we have it in the DataSource